ffs/scheduler/internal/trackstore/trackstore.go

```diff
@@ -18,41 +18,26 @@ import (
 // This store is used by the Scheduler background jobs that
 // repair or renew storage configurations.
 type Store struct {
-	lock        sync.Mutex
-	ds          datastore.Datastore
-	repairables map[cid.Cid]struct{}
-	renewables  map[cid.Cid]struct{}
+	lock sync.Mutex
+	ds   datastore.Datastore
 }
 
-type trackedStorageConfig struct {
+// TrackedStorageConfig has information about a StorageConfig from
+// an APIID which is repairable/renewable.
+type TrackedStorageConfig struct {
 	IID           ffs.APIID
 	StorageConfig ffs.StorageConfig
 }
 
-// New retruns a new Store.
-func New(ds datastore.Datastore) (*Store, error) {
-	s := &Store{
-		ds:          ds,
-		repairables: map[cid.Cid]struct{}{},
-		renewables:  map[cid.Cid]struct{}{},
-	}
-	if err := s.loadCaches(); err != nil {
-		return nil, fmt.Errorf("loading renewable/repairable caches: %s", err)
-	}
-	return s, nil
+// TrackedCid contains tracked storage configs for a Cid.
+type TrackedCid struct {
+	Cid     cid.Cid
+	Tracked []TrackedStorageConfig
 }
```

Comment:
I guess I'm not familiar with the implementation of repair/renew, but just curious why storing a list per cid is better than using a `instanceId/cid` key with a `StorageConfig` value, avoiding that complexity you mentioned.

Reply:
Yeah, this was mostly for avoiding changing the original keyspace namespace structure since at that time I didn't have all the migration library stuff that did afterward; so thought originally was a better idea. After having the migration library, looks like might be a better idea.

Anyway, I don't like the idea of this `Trackstore` in general. I believe in the future we should take it out, and maybe only save the `iid`, and find a reasonable way to query the corresponding API for the `StorageConfig`. Since here we're kind of duplicating that info. Still feels weird since that kind of violates the Scheduler knowing about a higher-level concept such as `API` but dunno.